utils_cv/detection/dataset.py

```diff
@@ -231,17 +235,14 @@ def split_train_test(
         Return
             A training and testing dataset in that order
         """
-        # TODO Is it possible to make these lines in split_train_test() a bit
-        # more intuitive?
-
         test_num = math.floor(len(self) * (1 - train_pct))
         indices = torch.randperm(len(self)).tolist()
 
-        self.transforms = get_transform(train=True)
-        train = Subset(self, indices[test_num:])
+        train = copy.deepcopy(Subset(self, indices[test_num:]))
```

Comment:
IS the copy.deepcopy() required? I my code it seems to be working without

Reply:
oh really? i thought i had tested it without and it didn't work... let me verify

Reply:
yeah, it does __not__ work without the deepcopy

Reply:
Strange, let's use the copy then, though for me it seems to be working withouth. By the way, this class takes as input "self.transforms" and looks like this input is not used anywhere @jiata

Reply:
thats a good point, actually its misleading even. let me do a refactor...